rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -189,23 +190,7 @@ func NewLayer2Relayer(ctx context.Context, l2Client *ethclient.Client, db *gorm.
 		return nil, fmt.Errorf("invalid service type for l2_relayer: %v", serviceType)
 	}
 
-	// pick and validate submission strategy
-	windowSec := uint64(cfg.BatchSubmission.TimeoutSec)
-	strategy, ok := bestParams[windowSec]
-	if !ok {
-		return nil, fmt.Errorf(
-			"unsupported BatchSubmission.TimeoutSec: %d (must be one of %v)",
-			windowSec, maps.Keys(bestParams),
-		)
-	}
-	layer2Relayer.batchStrategy = strategy
-
-	latest, err := layer2Relayer.l1BlockOrm.GetLatestL1BlockHeight(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get latest L1 block height: %v", err)
-	}
-	layer2Relayer.lastFetchedBlock = latest - uint64(layer2Relayer.cfg.BatchSubmission.TimeoutSec)/12 // start ~window seconds ago
-	if _, err = layer2Relayer.fetchBlobFeeHistory(uint64(layer2Relayer.cfg.BatchSubmission.TimeoutSec)); err != nil {
+	if _, err := layer2Relayer.fetchBlobFeeHistory(uint64(layer2Relayer.cfg.BatchSubmission.TimeoutSec)); err != nil {
```

Comment:
This is exactly called as is in `skipSubmitByFee`. I think we can just remove it from the constructor no?

Reply:
Yes, it can be removed. It would just prepare the big chunk of the history on construction instead of during the loop. Otherwise it is fine.